src/api/app/controllers/webui/patchinfo_controller.rb

```diff
@@ -158,6 +159,7 @@ def save
             Package.verify_file!(@package, '_patchinfo', xml)
           rescue APIException => e
             flash[:error] = "patchinfo is invalid: #{e.message}"
+            render action: 'edit_patchinfo', project: @project, package: @package
```

Comment:
This render is needed if you call the action directly. In the UI is a previous step that makes this rescue "never" visited.